From d5ffea7cf35cea42f4dc5a605cc15e117e3f18ce Mon Sep 17 00:00:00 2001
From: Marc Kleine-Budde <mkl@blackshift.org>
Date: Sat, 12 Sep 2015 17:29:59 +0200
Subject: [PATCH 087/198] net: rtl8192cu: Fix off-by-one warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit 830773130aa1b776eb0674942d630e0ebd81299d from
https://github.com/raspberrypi/linux.git

../drivers/net/wireless/realtek/rtl8192cu/hal/rtl8192c/rtl8192c_rf6052.c: In function ‘PHY_RFShadowRefresh’:
../drivers/net/wireless/realtek/rtl8192cu/hal/rtl8192c/rtl8192c_rf6052.c:1020:37: warning: iteration 63 invokes undefined behavior [-Waggressive-loop-optimizations]
    RF_Shadow[eRFPath][Offset].Value = 0;
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
../drivers/net/wireless/realtek/rtl8192cu/hal/rtl8192c/rtl8192c_rf6052.c:1018:3: note: within this loop
   for (Offset = 0; Offset <= RF6052_MAX_REG; Offset++)
   ^~~

Signed-off-by: Marc Kleine-Budde <mkl@blackshift.org>
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../wireless/realtek/rtl8192cu/hal/rtl8192c/rtl8192c_rf6052.c   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/realtek/rtl8192cu/hal/rtl8192c/rtl8192c_rf6052.c b/drivers/net/wireless/realtek/rtl8192cu/hal/rtl8192c/rtl8192c_rf6052.c
index cf46969379a8..5995a4b85dc5 100755
--- a/drivers/net/wireless/realtek/rtl8192cu/hal/rtl8192c/rtl8192c_rf6052.c
+++ b/drivers/net/wireless/realtek/rtl8192cu/hal/rtl8192c/rtl8192c_rf6052.c
@@ -1015,7 +1015,7 @@ PHY_RFShadowRefresh(
 
 	for (eRFPath = 0; eRFPath < RF6052_MAX_PATH; eRFPath++)
 	{
-		for (Offset = 0; Offset <= RF6052_MAX_REG; Offset++)
+		for (Offset = 0; Offset < RF6052_MAX_REG; Offset++)
 		{
 			RF_Shadow[eRFPath][Offset].Value = 0;
 			RF_Shadow[eRFPath][Offset].Compare = _FALSE;
-- 
2.17.1

